data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/oidc/TokenProvider.java

```diff
@@ -15,30 +15,51 @@
  */
 package dev.knative.eventing.kafka.broker.core.oidc;
 
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
 import dev.knative.eventing.kafka.broker.core.NamespacedName;
 import io.fabric8.kubernetes.api.model.authentication.TokenRequest;
 import io.fabric8.kubernetes.api.model.authentication.TokenRequestBuilder;
-import io.fabric8.kubernetes.client.Config;
-import io.fabric8.kubernetes.client.ConfigBuilder;
 import io.fabric8.kubernetes.client.KubernetesClient;
-import io.fabric8.kubernetes.client.KubernetesClientBuilder;
+
+import java.util.concurrent.TimeUnit;
 
 public class TokenProvider {
 
     private final KubernetesClient kubernetesClient;
+    private final Cache<String, String> tokenCache;
+
+    public TokenProvider(KubernetesClient kubernetesClient) {
+        this.kubernetesClient = kubernetesClient;
+
+        this.tokenCache = CacheBuilder.newBuilder()
+                .expireAfterWrite(1, TimeUnit.HOURS) // 1 hour expiration after write
```

Comment:
Can we use a constant here and align with line 62 (`.withExpirationSeconds(3600L)`). Also we should add some buffer to expire the tokens a bit earlier to prevent clock-screw.